### DISCUSSION TITLE
Gain calibration windows - instructions

### DISCUSSION THREAD

**Discussion by agentpatience** [OpenHantek/OpenHantek6022#315] [General]
I lost the instructions for gain offset calibration for Windows. can you help? I cant seem to find the proper link. 

Can someone please advise?

Thanks,

**Comment by agentpatience** [comment]
![image0](https://user-images.githubusercontent.com/36264810/178109308-4c9a2b2a-35c0-48e3-9696-44b140966e8c.jpeg)

Its off by 700mv?! I adjusted PSU with my DMM beforehand.

**Reply by Ho-Ro** [reply]
Please provide more info about the scope settings and the applied signal.

Hint - you do not need to make a photo, you can take a screenshot right out of the program in menu **Export**.
![image](https://user-images.githubusercontent.com/12542359/178117929-efa877ae-1634-422c-9500-134e9ce9871c.png)

**Comment by agentpatience** [comment]
Hi HoRo I made a couple videos that hopefully show a couple issues a little more clearly. I’m trying to isolate user error as well. I own a DSO2090 which has a higher input voltage and allows me to not need to attenuate the probes over 5 volts. It also seems generally more accurate.

The videos are here for your review:
https://youtu.be/YTEjbptG8A4
https://youtu.be/AJNNJijPpxA

**Reply by Ho-Ro** [reply]
Some general remarks to your findings: 
- The maximum input voltage of the Hantek6022 is about 4.5 V, if you measure higher voltages you need the 1:10 probe.
- If you are measuring sources with high internal resistance, then loading them with the internal resistance of the measuring instrument will reduce the measured voltage oscilloscopes typically have 1 MOhm when connected directly (it also adds capacitive load of about 25pF), with 1:10 probe it becomes 10 MOhm (plus about 5pF). DMMs typically have a much higher input resistance, which also makes the cut-off frequency much lower than that of an oscilloscope.

**Comment by agentpatience** [comment]
Ok, here is your screenshot with one of the stock probes. Same setup as in the video as you can see the stock probe is worse. This is x10 mode on the probe and on the software. The DC source is 9vDC.

![image](https://user-images.githubusercontent.com/36264810/178125597-fa5bf0b1-1e4f-4394-b5ab-ff7317ecd0af.png)

**Comment by Ho-Ro** [comment]
There exist two reasons for wrong voltage reading - offset and gain. You must correct both and do it in the right sequence. This can be done all-in-one by using the [python tool](https://github.com/Ho-Ro/Hantek6022API#create-calibration-values-for-openhantek) (this link describes also the reason for the relative big error) or do it step by step:

1. Correct the offset error, this can be done directly with OH6022 - menu `Oscilloscope/Calibrate Offset`  and follow the instructions.
2. Now you can correct the gain error, just apply a defined voltage to both channels and write down the measured DC values (do it at least for 1V/div, 500mV/div, 200mV/div and 100 mV/div, the three upper ranges 1, 2, 5V have the same gain, as well as the three lower ranges 100, 50, 20mV). The correction value calculates as corr = Uin/Udisplay. Put these correction values into a calibration file, for Windows the path is `%APPDATA%\OpenHantek\DSO-6022BE_NNNNNNNNNNNN_calibration.ini`, NNN... is the serial number of your scope, see also the [comment in source code](https://github.com/OpenHantek/OpenHantek6022/blob/e5f0e4a2b25f9738cd66afe90a1ada6440ce51e5/openhantek/src/hantekdso/hantekdsocontrol.cpp#L459-L462):
```
[eeprom]
replace_eeprom=false

[gain]
ch0\1000mV=1
ch0\100mV=1
ch0\2000mV=1
ch0\200mV=1
ch0\20mV=1
ch0\5000mV=1
ch0\500mV=1
ch0\50mV=1
ch1\1000mV=1
ch1\100mV=1
ch1\2000mV=1
ch1\200mV=1
ch1\20mV=1
ch1\5000mV=1
ch1\500mV=1
ch1\50mV=1

[offset]
ch0\1000mV=0
ch0\100mV=0
ch0\2000mV=0
ch0\200mV=0
ch0\20mV=0
ch0\5000mV=0
ch0\500mV=0
ch0\50mV=0
ch1\1000mV=0
ch1\100mV=0
ch1\2000mV=0
ch1\200mV=0
ch1\20mV=0
ch1\5000mV=0
ch1\500mV=0
ch1\50mV=0
```

**Comment by FroggyCorp** [comment]
Hi,

I didn't use my hantek for a long time because i got some gain/offset problem, so today i try with the last nightly version of openhantek6022 to see if there is some difference. I still get constant 0.05V diff per 1V (1v -> 1.05v, 2v -> 2.10v, ...). It's not big but bit borring.

I connect directly the + and - of CH1 & CH2
I try the "oscilloscope/calibration offset" with no result. It write nothing nowhere.
I try the python script (with admin shell), it write the DSO-???.ini file, but its a "blank" file as your example.

Do you have any idea to solve it ?
It's look like perfect 5% difference.

I'm using Windows 10, OpenHantek6022 3.3.12-22 and a 6022BE (FW0210).

Thx in advance,

**Reply by Ho-Ro** [reply]
Did you try the proposed solution from above and wrote your calibration values in the config file `%APPDATA%\OpenHantek\DSO-6022BE_NNNNNNNNNNNN_calibration.ini`?

```
...
[gain]
ch0\1000mV=0.95
ch0\100mV=0.95
...
```

**Comment by FroggyCorp** [comment]
Hi,

Yep, that works, thanks.
